### ISSUE TITLE
Portable version?

### ISSUE THREAD

**Issue by jchodera** [isayev/ASE_ANI#19]
Just curious why you're distributing binary libraries instead of a platform-portable version here. Is the actual code that could be targeted to multiple systems available in another repo?

**Comment by isayev** [comment]
John:
Thanks for comments. We will keep this in mind. Let me know if yoy would
like to play with ANI in a Docker container. We have a prototype version.

**Comment by jchodera** [comment]
A docker container at least enables reproducibility and portability, though not understanding and innovation. I think a docker container would still be useful for us to play around with! Do you have one on dockerhub?

**Comment by dgasmith** [comment]
Could these custom models be transferred over to standard TF/Keras/PyTouch backends, would make transferability and accessibility an order of magnitude higher.

**Comment by jchodera** [comment]
Looks like someone has already done this!

https://github.com/proteneer/khan

**Comment by roitberg** [comment]
HI guys. We have a beta version in pytorch and will release it soon. The khan version is not quite ANI. The Deepchem also is not.
The basis are there, but the devil is in the details of activation functions, epochs, gradients, etc etc

**Comment by jchodera** [comment]
> The basis are there, but the devil is in the details of activation functions, epochs, gradients, etc etc

So the paper is useless without releasing all the code, since it's an incomplete description of what you actually did? :)

**Comment by jchodera** [comment]
Sounds like a great argument for releasing the code!

**Comment by roitberg** [comment]
It is actually the COMPLETE opposite. In the case of Deepchem for instance, they implemented whatever the hell they felt like, even when the paper was VERY clear about our choices. They published a paper showing that ANI was bad, and it turned to have been their own mistakes. That is what I meant. Take khan or deepchem, read our paper carefully and implement OUR choices there. Then the comparison is fair

**Comment by jchodera** [comment]
Got it! Thanks for the clarification!

**Comment by isayev** [comment]
@jchodera paper has a *complete* and *honest* description of what we did. Not every one pay attention to full technical details in SI or simply made their own informed decision on those parameters.  

Like, wtf these guys are crazy to use gaussian activation function, let me use laters greatest ReLU etc.

**Comment by jchodera** [comment]
Awesome. Looking forward to the `pytorch` implementation then!

**Comment by isayev** [comment]
@jchodera @proteneer @dgasmith  Pytorch ANI is now available: https://github.com/aiqm/torchani
